reactive/kubernetes_master.py

```diff
@@ -1869,14 +1870,13 @@ def token_generator(length=32):
 
 @retry(times=3, delay_secs=1)
 def get_pods(namespace='default'):
-    cmd = [
-        'kubectl', 'get', 'po',
-        '-n', namespace,
-        '-o', 'json',
-        '--request-timeout', '10s'
-    ]
     try:
-        output = check_output(cmd).decode('utf-8')
+        output = kubectl(
+            'get', 'po',
+            '-n', namespace,
+            '-o', 'json',
+            '--request-timeout', '10s'
+        ).decode('UTF-8')
```

Comment:
I don't believe the decodes are necessary here. Are you sure you need them?

Reply:
I don't see them in the kubectl helper, but I've used them to capture output without the decode in another place. We don't capture output in another spot in master or worker though.

Reply:
Looks like it's not necessary in python 3.6

```
$ python3
Python 3.6.7 (default, Oct 31 2018, 13:29:56) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> import subprocess
>>> output = subprocess.check_output(['cat', 'some.json'])
>>> json.loads(output)
{'foo': 'bar'}
```

But it is necessary in python 3.5:
```
$ python3.5
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> import subprocess
>>> output = subprocess.check_output(['cat', 'some.json'])
>>> json.loads(output)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
>>> json.loads(output.decode('UTF-8'))
{'foo': 'bar'}
```

I don't have a full understanding of which python versions are used by the charms in what situations. My preference would be to play it safe and support both.